From e609f49f172b98bccace43bae603d7cd4b861d54 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jozef=20=C4=8C=C3=AD=C5=BE?= <jozef.ciz.ml@gmail.com>
Date: Sat, 11 Feb 2023 13:39:02 +0100
Subject: [PATCH] Add InitWithFile method to ZepEditor

A method to initialize editor with file but not change working directory
was missing.
---
 include/zep/editor.h | 1 +
 src/editor.cpp       | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/include/zep/editor.h b/include/zep/editor.h
index 5bee5e6d..8233d845 100644
--- a/include/zep/editor.h
+++ b/include/zep/editor.h
@@ -264,6 +264,7 @@ public:
 
     void Reset();
     ZepBuffer* InitWithFileOrDir(const std::string& str);
+    ZepBuffer* InitWithFile(const std::string& str);
     ZepBuffer* InitWithText(const std::string& strName, const std::string& strText);
 
     ZepMode* GetGlobalMode();
diff --git a/src/editor.cpp b/src/editor.cpp
index 166c3a16..99bc0d27 100644
--- a/src/editor.cpp
+++ b/src/editor.cpp
@@ -451,6 +451,12 @@ ZepBuffer* ZepEditor::InitWithFileOrDir(const std::string& str)
         }
     }
 
+    return InitWithFile(str);
+}
+
+ZepBuffer* ZepEditor::InitWithFile(const std::string& str)
+{
+    ZepPath startPath(str);
     // Get a buffer for the start file; even if the path is not valid; it can be created but not saved
     auto pFileBuffer = GetFileBuffer(startPath);
     auto pTab = EnsureTab();
-- 
2.39.1

